src/projection.jl

```diff
@@ -210,6 +210,11 @@ function (project::ProjectTo{AbstractArray})(dx::AbstractArray{S,M}) where {S,M}
     return dz
 end
 
+# Trivial case -- won't collapse Any[NoTangent(), NoTangent()]
+function (project::ProjectTo{AbstractArray})(dx::AbstractArray{<:AbstractZero})
+    return NoTangent()
+end
```

Comment:
This fits on one line
```suggestion
(project::ProjectTo{AbstractArray})(dx::AbstractArray{<:AbstractZero}) = NoTangent()
```